website/src/utils/search.ts

```diff
@@ -101,155 +80,141 @@ export const getColumnVisibilitiesFromQuery = (schema: Schema, state: Record<str
-    const result: MetadataFilter[] = [];
-    for (const field of metadataSchema) {
-        if (field.rangeOverlapSearch) {
-            const fieldGroupProps = {
-                fieldGroup: field.rangeOverlapSearch.rangeName,
-                fieldGroupDisplayName: field.rangeOverlapSearch.rangeDisplayName,
-            };
-            result.push({
-                ...field,
-                ...fieldGroupProps,
-                name: `${field.name}From`,
-                label: 'From',
-            });
-            result.push({
-                ...field,
-                ...fieldGroupProps,
-                name: `${field.name}To`,
-                label: 'To',
-            });
-        } else if (field.rangeSearch === true) {
-            const fromField = {
-                ...field,
-                name: `${field.name}From`,
-                label: 'From',
-                fieldGroup: field.name,
-                fieldGroupDisplayName: field.displayName ?? sentenceCase(field.name),
-            };
-            const toField = {
-                ...field,
-                name: `${field.name}To`,
-                label: 'To',
-                fieldGroup: field.name,
-                fieldGroupDisplayName: field.displayName ?? sentenceCase(field.name),
-            };
-            result.push(fromField);
-            result.push(toField);
-        } else {
-            result.push(field);
-        }
-    }
-    return result;
-};
-
-export type ConsolidatedMetadataFilters = (MetadataFilter | GroupedMetadataFilter)[];
-
 /**
- * Take a list of MetadataFilters and return a new list where filters that belong to a group
- * are grouped together into GroupedMetadataFilters.
+ * Derives from the Metadata schema. For some metadata fields, they are expanded into multiple
+ * (grouped) filters.
  */
-export const consolidateGroupedFields = (filters: MetadataFilter[]): ConsolidatedMetadataFilters => {
-    const fieldList: (MetadataFilter | GroupedMetadataFilter)[] = [];
-    const groupsMap = new Map<string, GroupedMetadataFilter>();
-
-    for (const filter of filters) {
-        if (filter.fieldGroup !== undefined) {
-            if (!groupsMap.has(filter.fieldGroup)) {
-                const fieldForGroup: GroupedMetadataFilter = {
-                    name: filter.fieldGroup,
-                    groupedFields: [],
-                    type: filter.type,
-                    grouped: true,
-                    displayName: filter.fieldGroupDisplayName,
-                    label: filter.label,
-                    initiallyVisible: filter.initiallyVisible,
-                };
-                fieldList.push(fieldForGroup);
-                groupsMap.set(filter.fieldGroup, fieldForGroup);
+export class FilterSchema {
+    public readonly filters: (MetadataFilter | GroupedMetadataFilter)[] = [];
+
+    constructor(metadataSchema: Metadata[]) {
+        // expand metadata fields for ranges and range overlap search
```

Comment:
These things - expanding date ranges to a series of fields with a group property, and grouping anything with a group property together, used to be their own functions. IMO that made things easier to understand. We could still do it in some way with this design I presume?

Reply:
Yes we could, and I will put them back into their own functions, it's a good idea. at first I thought (and tried) to merge it all into one shorter function to make it easier, but that didn't work out.

Reply:
[`3d75165` (#3883)](https://github.com/loculus-project/loculus/pull/3883/commits/3d75165e8df56b405605a3f24837cf23413f2321)